Tools/DataStorage/DataStorage.cs

```diff
@@ -103,6 +229,174 @@ public Class GetClass(string classId)
+                    }
+
+                    reader.Close();
+                }
+
+                return list;
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("There's problem with you connection!\n" + ex.Message);
+            }
+            finally
+            {
+                myConn?.Close();
+            }
+
+            return list;
+
+        }
+
+        public List<Teacher> GetTeachers(string year)
+        {
+            string sql = "SELECT ht.tab_number, ht.h_name, ht.patronymic, c.class_id, c.number, c.letter, c.st_year, ht.surname" +
+                         " FROM (head_teachers ht INNER JOIN head_teachers_classes htc ON ht.tab_number=htc.tab_number) " +
+                         "INNER JOIN classes c ON c.class_id = htc.class_id "+
+                         $"WHERE c.st_year='{year}'";
+
+            List<Teacher> res = new List<Teacher>();
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+
+                using (SqlCommand command = new SqlCommand(sql, myConn))
+                {
+                    var reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        Teacher cur = new Teacher(reader.GetString(0));
+                        cur.HName = reader.GetString(1);
+                        cur.Patronymiс = reader.GetString(2);
+                        cur.Surname = reader.GetString(7);
+                        Class c = new Class(reader.GetInt64(3).ToString());
+                        c.Number = reader.GetString(4);
+                        c.Letter = reader.GetString(5);
+                        c.StYear = reader.GetString(6);
+                        cur.Class = c;
+                   
+                        res.Add(cur);
+                    }
+
+                    reader.Close();
+                }
+
+                return res;
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show("There's problem with you connection!\n" + ex.Message);
+            }
+            finally
+            {
+                myConn?.Close();
+            }
+
+            return res;
+        }
+
+        public List<User> GetUsers()
+        {
+            string sql = "SELECT password, login, rights" +
+                                  " FROM [user] ";
+
+            List<User> res = new List<User>();
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+
+                using (SqlCommand command = new SqlCommand(sql, myConn))
+                {
+                    var reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        User cur = new User(reader.GetString(1), reader.GetString(0), reader.GetString(2));
```

Comment:
My personal advice, use only c-tors by default and assign property values using setters. The code is more usable then. Like:
User u = new User();
u.Login = "bla";
u.Password = "blo"
u.AccessType = "blu"
and so on.

Reply:
hm, don't think it is so important, but maybe you are 
right